### ISSUE TITLE
Websocket renew subscription failure

### ISSUE THREAD

**Issue by KatunaNorbert** [oceanprotocol/pdr-websocket#48]
Websocket fails when tries to renew the subscription and has to be manually restarted

**Comment by trentmc** [comment]
We should take steps to make this more reliable, now that more eyes are on predictoor

Perhaps simply re-starting every 10min? @trizin

**Comment by KatunaNorbert** [comment]
If we are restarting it at the same time each day then the cron is probably trying to restart too early and the subscription is not expired yet. What we could do is keep track of the time it was restarted last time and add +5min to the next time so it gives some extra time for the subscription to expire before restarting.

**Comment by KatunaNorbert** [comment]
Another solution might be switching the **rpc** from `wss://sapphire.oasis.io/ws` to the `Chainstack` rpc which is more reliable

**Comment by KatunaNorbert** [comment]
It does but it's not working.. was spending some time with @kdetry on fixing it but didn't find a solution so decided to rely on the cron approach. We could take another look again if the cron approach it's not working.

**Comment by idiom-bytes** [comment]
```
It does but it's not working.. was spending some time with @kdetry on fixing it but didn't find a solution so decided to rely on the cron approach. We could take another look again if the cron approach it's not working.
```
Whether the cron job is working or not, having an unstable backend service isn't good. I agree with berkay that we need to address this at the service-level (code), not system-level (cron).

```
It fails sometimes when trying to renew the subscription
```
Great, what else?

Is there any logging?
Do we understand the error that's happening when trying to renew the subscription?

Will try/catches help?
Will retries help?

**Can we expand/document this on github/issue with more information** on the problem and what we've done to solve "pdr-websocket fails to renew subscription"? Please provide context in ticket.

**Comment by KatunaNorbert** [comment]
I don't think that PR fixes the issue. As far as I know it's not solved that's why it wasn't merged.
The issue I encountered is either the subscription has ended and we try to get the prediction or has not ended and we try to renew it. To sum it up the subscription renewal is well timed.

**Comment by trentmc** [comment]
> I don't think that PR fixes the issue. As far as I know it's not solved that's why it wasn't merged.

Correct, this is what Mustafa said a few days ago (in slack I think?)
